### PR TITLE
feat: DISCO-1805 - Split pytest into groups

### PR DESCRIPTION
**Pull request by shafqatfarhan**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -10,6 +10,7 @@ testpaths = course_discovery/apps
 deps =
     django32: -r requirements/django.txt
     -r{toxinidir}/requirements/local.txt
+    pytest-split == 0.7.0
```
Comment (waheedahmed) [inline tox.ini]: Add it into `requirements/local.txt` instead.

File: .gitignore [hunk 2]
```diff
@@ -97,4 +97,5 @@ webpack-stats.json
 .vscode
 
 # Emacs
-*~
\ No newline at end of file
+*~
+.test_durations
```
Comment (waheedahmed) [inline .gitignore]: Add a new line at the end of the file.

File: .test_durations [hunk 3]
```diff
@@ -0,0 +1,1576 @@
...
+    "course_discovery/apps/edx_catalog_extensions/api/v1/tests/test_views.py::ProgramFixtureViewTests::test_404_nonexistent": 8.619234784000582,
+    "course_discovery/apps/edx_catalog_extensions/api/v1/tests/test_views.py::ProgramFixtureViewTests::test_422_too_many_programs": 15.393095580999216,
+    "course_discovery/apps/edx_catalog_extensions/api/v1/tests/test_views.py::ProgramFixtureViewTests::test_exception_failed_load_objects": 8.974441325000043,
+    "course_discovery/apps/edx_elasticsearch_dsl_extensions/distinct_counts/tests/test_query.py::TestDistinctCountsElasticsearchQueryWrapper::test_build_search_kwargs_does_not_include_facet_clause": 7.595899505000489,
+    "course_discovery/apps/edx_elasticsearch_dsl_extensions/distinct_counts/tests/test_query.py::TestDistinctCountsElasticsearchQueryWrapper::test_search_raises_when_called_with_date_facet": 6.602929790999951,
+    "course_discovery/apps/edx_elasticsearch_dsl_extensions/distinct_counts/tests/test_query.py::TestDistinctCountsElasticsearchQueryWrapper::test_search_raises_when_called_with_unsupported_field_facet_option": 6.9457686160003504,
+    "course_discovery/apps/edx_elasticsearch_dsl_extensions/distinct_c
```
Comment (waheedahmed) [inline .test_durations]: Add a new line at the end of the file.

**Review by waheedahmed — COMMENTED** [review]
Few minor comments otherwise looks good.

**Review by zainab-amir — COMMENTED** [review]
Is the file .test_durations meant to be added or was it added by mistake?

**Comment by shafqatfarhan** [comment]
> Is the file .test_durations meant to be added or was it added by mistake?

Yes, it is required for splitting algorithm that pytest-split uses

File: tox.ini (hunk 1 above)
Comment (shafqatfarhan) [reply tox.ini]: done

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**